source/postprocess/viscous_dissipation_per_composition.cc

```diff
@@ -0,0 +1,177 @@
+                        const SymmetricTensor<2, dim> deviatoric_strain_rate =
+                          (this->get_material_model().is_compressible()
+                           ? in.strain_rate[q] - 1. / 3. * trace(in.strain_rate[q]) * unit_symmetric_tensor<dim>()
+                           : in.strain_rate[q]);
+                        local_dissipation_integral[c] += (out.viscosities[q] * deviatoric_strain_rate) *
+                                                         deviatoric_strain_rate * fe_values.JxW(q);
+                      }
+                  }
+              }
+          }
+
+      std::vector<double> viscous_dissipation (local_dissipation_integral.size());
+      Utilities::MPI::sum (local_dissipation_integral, this->get_mpi_communicator(), viscous_dissipation);
+
+      for (unsigned int c = 0; c < this->n_compositional_fields(); ++c)
+        {
+          if (this->convert_output_to_years() == true)
+            {
+              statistics.add_value("Viscous dissipation (J/year/m) for composition " + this->introspection().name_for_compositional_index(c),
+                                   viscous_dissipation[c] * year_in_seconds);
+
+              // also make sure that the other columns filled by the this object
+              // all show up with sufficient accuracy and in scientific notation
+              {
+                const std::string columns[] = { "Viscous dissipation (J/year/m) for composition " + this->introspection().name_for_compositional_index(c)};
+                for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+                  {
+                    statistics.set_precision (columns[i], 8);
+                    statistics.set_scientific (columns[i], true);
+                  }
+              }
+            }
+          else
+            {
+              statistics.add_value ("Viscous dissipation (W/m) for composition " + this->introspection().name_for_compositional_index(c),
+                                    viscous_dissipation[c]);
+
+              // also make sure that the other columns filled by the this object
+              // all show up with sufficient accuracy and in scientific notation
+              {
+                const std::string columns[] = { "Viscous dissipation (W/m) for composition " + this->introspection().name_for_compositional_index(c)};
+                for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+                  {
+                    statistics.set_precision (columns[i], 8);
+                    statistics.set_scientific (columns[i], true);
+                  }
+              }
+            }
+        }
+
+      std::ostringstream output;
+      output.precision(4);
+      for (unsigned int c = 0; c < this->n_compositional_fields(); ++c)
+        {
+          if (this->convert_output_to_years() == true)
+            output << viscous_dissipation[c] * year_in_seconds
+                   << " J/year/m";
```

Comment:
I would not bother converting the years here, it is confusing. Nobody uses `J/year`, everyone uses `W (J/s)`.